### DISCUSSION TITLE
Why not add ftp functionality?  I think that's useful

### DISCUSSION THREAD

**Discussion by yys-bury** [cesanta/mongoose#2319] [Ask questions, get answers] [answered]
- My goal is: _describe_
- My actions were: _describe_
- My expectation was: _describe_
- The result I saw: _provide logs or description_
- My question is: _your question_

# Environment

- mongoose version: _X.Y_
- Compiler/IDE and SDK: ZZZ version _X.Y_
- Target hardware/board: _?_
- Connectivity chip/module: _?_
- Target RTOS/OS (if applicable): _?_

**Comment by cpq** [comment, marked as answer]
@yys-bury we haven't been asked for FTP even once. File uploads, required for firmware / config updates, are easily done over HTTP. There is no point in introducing a whole protocol for that.

**Comment by hywing** [comment]
I think ftp protocol is too big for mongoose, use http for file transportation instead.
